azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/Certificate.java

```diff
@@ -0,0 +1,594 @@
+         return new AutoValue_Certificate_LifetimeActionTrigger(daysBeforeExpiry, lifetimePercentage);
+      }
+   }
+
+   @AutoValue
+   public abstract static class LifetimeActionAction {
+      public abstract String actionType();
+
+      @SerializedNames({"action_type"})
+      public static LifetimeActionAction create(final String actionType) {
+         return new AutoValue_Certificate_LifetimeActionAction(actionType);
+      }
+   }
+
+   @AutoValue
+   public abstract static class LifetimeAction {
+      public abstract LifetimeActionAction action();
+
+      public abstract LifetimeActionTrigger trigger();
+
+      @SerializedNames({"action", "trigger"})
+      public static LifetimeAction create(final LifetimeActionAction action,
+                                          final LifetimeActionTrigger trigger) {
+         return new AutoValue_Certificate_LifetimeAction(action, trigger);
+      }
+   }
+
+   @AutoValue
+   public abstract static class SecretProperties {
+      public abstract String contentType();
+
+      @SerializedNames({"contentType"})
+      public static SecretProperties create(final String contentType) {
+         return new AutoValue_Certificate_SecretProperties(contentType);
+      }
+   }
+
+   @AutoValue
+   public abstract static class SubjectAlternativeNames {
+      public abstract List<String> dnsNames();
+
+      public abstract List<String> emails();
+
+      public abstract List<String> upns();
+
+      @SerializedNames({"dns_names", "emails", "upns"})
+      public static SubjectAlternativeNames create(final List<String> dnsNames,
+                                                   final List<String> emails,
+                                                   final List<String> upns) {
+         return new AutoValue_Certificate_SubjectAlternativeNames(dnsNames, emails, upns);
+      }
+   }
+
+   @AutoValue
+   public abstract static class X509CertificateProperties {
+      @Nullable
+      public abstract List<String> enhancedKeyUsage();
+
+      @Nullable
+      public abstract List<String> keyUsage();
+
+      @Nullable
+      public abstract SubjectAlternativeNames subjectAltNames();
+
+      @Nullable
+      public abstract String subject();
+
+      public abstract int validityMonths();
+
+      @SerializedNames({"ekus", "key_usage", "sans", "subject", "validity_months"})
+      public static X509CertificateProperties create(final List<String> enhancedKeyUsage,
+                                                     final List<String> keyUsage,
+                                                     final SubjectAlternativeNames subjectAltNames,
+                                                     final String subject,
+                                                     final int validityMonths) {
+         return new AutoValue_Certificate_X509CertificateProperties(enhancedKeyUsage, keyUsage, subjectAltNames, subject, validityMonths);
```

Comment:
Enforce immutability in all lists when present. Is there a reason to keep these lists null, or could we better default to empty lists, to avoid the null-collection anti-pattern?